src/community/schemaless-features/schemaless-mongo/src/main/java/org/geoserver/schemalessfeatures/mongodb/data/MongoComplexContentDataAccess.java

```diff
@@ -122,4 +121,10 @@ protected FilterCapabilities createFilterCapabilities() {
     public void dispose() {
         client.close();
     }
+
+    private Set<String> getCollectionNames() {
+        Set<String> collectionNames = new LinkedHashSet<>();
+        database.listCollectionNames().forEach(n -> collectionNames.add(n));
```

Comment:
If ```database.listCollectionNames()``` is a Collection, you can create the Set using the constructor directly: ```new LinkedHashSet(database.listCollectionNames());```

Reply:
the listCollectionNames() returns a  MongoIterable object so iteration is needed.